### ISSUE TITLE
Implement `exclusive` Property for Cfgu Declaration

### ISSUE THREAD

**Issue by rannn505** [configu/configu#241]
**Description**
Introduce an `exclusive` property to the Cfgu declaration. Presented as an array of strings, this attribute will ensure that the designated key cannot coexist with the keys listed in its array. In simpler terms, if one key is assigned a value, the other, as marked 'exclusive', can't be given one.

```json
{
  ...
  "CONFIG_KEY_1": {
    "type": "String",
    "exclusive": ["CONFIG_KEY_2"] // If CONFIG_KEY_2 is assigned, CONFIG_KEY_1 can't be.
  },
  "CONFIG_KEY_2": {
    "type": "String",
  },
  ...
} 
```

**Motivation**
This feature aims to bolster the flexibility and precision of the Cfgu declaration, ensuring clearer rules and minimizing potential conflicts between configuration keys.

**Additional context**

**Comment by trishanu-init** [comment]
@rannn505 Please assign me this issue
